components/mdi/Core/Src/main.c

```diff
@@ -303,6 +324,34 @@ int main(void)
     		}
     		HAL_CAN_AddTxMessage(&hcan, &TxHeader, TxData, TxMailbox);
     		SendSlowMessage = 0;
+
+        // MDI diagnostics 0x50F
+        TxHeader.StdId = MDI_DIAGNOSTICS_CAN_ID;
+        TxHeader.IDE = CAN_ID_STD;
+        TxHeader.DLC = 1;
+        TxData[0] = 0; // Clear TxData[0]
```

Comment:
The entire array should be cleared here, not just byte 0 (the rest will be garbage data)